macros/base/end_print.cfg

```diff
@@ -16,14 +16,12 @@ gcode:
     PARK
 
     {% if ercf_enabled %}
-        {% if unload|int == 1 %}
+        {% if unload|int == 99 and ercf_unload_on_end_print % }
             # unload filament and park into ercf
             ERCF_EJECT
-        {% elif unload|int == 0 %}
-        {% elif unload|int > 1 and ercf_unload_on_end_print %}
+        {% elif unload|int == 1 %}
             # unload filament and park into ercf
             ERCF_EJECT
-        {% else %}
         {% endif %}
     {% else %}
```

Comment:
I think it's too complicated.... As stated in another comment: slicer should be used in prioritary as it is for the other parameters.

Juste use something like :
``` 
# Set the parameter like that
{% set ercf_unload = params.ERCF_UNLOAD_AT_END|defaultprinter["gcode_macro _USER_VARIABLES"].ercf_unload_on_end_print)|int %}

# Then just use it in an easy way like that
[....]
{% if ercf_enabled and ercf_unload   %}
    ERCF_EJECT
{% else %}
[....]
```

Reply:
basically it does what you wanted, just not as clean as your solution :)